src/internal/app/app.go

```diff
@@ -169,6 +173,10 @@ func (a *application) registerChargers() error {
 	}
 
 	for _, charger := range chargers {
+		if a.ad.ThingByID(charger.ID) != nil {
```

Comment:
Please make the code inside the loop utilize `EnsureThings` method on the adapter.